### PR TITLE
[4.3] Basic docs for opentelemetry support

### PR DESCRIPTION
**Pull request by eldarnash**

### CODE REVIEW

File: content/headless/opentelemetry-authentication.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+---
+## `auth()` {#auth}
+
+You can set authentication in the `auth()` option of the `opentelemetry()` driver. By default, authentication is disabled (`auth(insecure())`).
+
+The following authentication methods are available in the `auth()` block:
+
+### `adc()` {#adc}
+
+[Application Default Credentials (ADC)](https://cloud.google.com/docs/authentication/application-default-credentials). This authentication method is only available for destination. It accepts the `target-service-account()` option, where you can list service accounts to match against when authenticating the server.
```
Comment (alltilla) [inline content/headless/opentelemetry-authentication.md]: The `target-service-account()` option is for the `alts()` auth mode.

File: content/docs/parsers/opentelemetry/_index.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+title: Parse OpenTelemetry messages
+linktitle: OpenTelemetry
+---
+
+By default, {{% param "product_name" %}} doesn't parse the fields of incoming OpenTelemetry messages into name-value pairs, but are only available for forwarding using the `opentelemetry()` destination. To parse the fields into name-value pairs, use the `opentelemetry()`.
```
Comment (alltilla) [inline content/docs/parsers/opentelemetry/_index.md]:
```suggestion
By default, {{% param "product_name" %}} doesn't parse the fields of incoming OpenTelemetry messages into name-value pairs, but are only available for forwarding using the `opentelemetry()` destination. To parse the fields into name-value pairs, use the `opentelemetry()` parser.
```

File: content/docs/parsers/opentelemetry/_index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    - The actual data is mapped into `.otel.metric.data.<type>.<...>`, for example, `.otel.metric.data.gauge.data_points.0.time_unix_nano`.
+- The fields of traces are mapped into `.otel.span.<...>`, for example, `.otel.span.name`, `.otel.span.trace_state`. Repeated fields have an index, for example, `.otel.span.events.5.time_unix_nano`.
+
+## Mapping data types
+
+String, bool, int64, double, and bytes values are mapped to their respective {{% param "product_name" %}} name-value type, for example, `.otel.resource.attributes.string_key` becomes a string value. <!-- FIXME add a link to the value-pairs docs -->
+
+The mapping of AnyValue type fields is limited.
+
+`ArrayValue` and `KeyValueList` types are stored serialized with protobuf type. Note that `protobuf` and `bytes` types are only available, unless explicitly type cast. In this case, the content is base64-encoded into the bytes content. For example, `bytes(${.otel.log.span_id})`. When using template functions, use `--include-bytes`, for example, `$(format-json .otel.* --include-bytes`.
```
Comment (alltilla) [inline content/docs/parsers/opentelemetry/_index.md]:
```suggestion
`ArrayValue` and `KeyValueList` types are stored serialized with protobuf type. Note that `protobuf` and `bytes` types are only available, unless explicitly type cast. For example, `bytes(${.otel.log.span_id})`. When using template functions, use `--include-bytes`, for example, `$(format-json .otel.* --include-bytes`. In the case of `$(format-json)`, the content is base64-encoded into the bytes content.
```

File: content/docs/parsers/opentelemetry/_index.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
+---
+title: Parse OpenTelemetry messages
+linktitle: OpenTelemetry
+---
+
+By default, {{% param "product_name" %}} doesn't parse the fields of incoming OpenTelemetry messages into name-value pairs, but are only available for forwarding using the `opentelemetry()` destination. To parse the fields into name-value pairs, use the `opentelemetry()`.
+
+The `opentelemetry()` parser parses the fields into name-value pairs starting with the `.otel.` prefix.
```
Comment (alltilla) [inline content/docs/parsers/opentelemetry/_index.md:8]: We could link to the `.proto` files for more info about the parsed fields: https://github.com/open-telemetry/opentelemetry-proto/tree/v0.20.0/opentelemetry/proto

File: content/headless/opentelemetry-authentication.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  };
+```
+
+### `insecure()` {#insecure}
+
+This is the default method, authentication is disabled (`auth(insecure())`).
+
+### `tls()` {#tls}
+
+<!-- FIXME xinclude these from the other tls blocks -->
```
Comment (alltilla) [inline content/headless/opentelemetry-authentication.md:32]: [note]

The `opentelemetry()` drivers have a completely different `tls()` block implementation from the `network()` or `http()` drivers. Currently it has most of the features, but it is not guaranteed to have 100% feature parity.

Comment (alltilla) [reply content/headless/opentelemetry-authentication.md:32]: @eldarnash Oh I meant to write this note to you only, not to put it to the docs, too, however it is okay to have it there. I will make sure to differentiate between a doc note and a note to the author when writing a comment. Sorry!

**Review by alltilla — APPROVED** [review]

**Comment by eldarnash** [comment]
Moved to the core-docs repo and published from there

**Pull request closed without merging**